test/contracts/root-element.tsx

```diff
@@ -0,0 +1,42 @@
+import {ClientRenderer, expect, RenderingContext, simulate, sinon, waitFor} from 'test-drive-react';
+import React = require('react');
+import {findDOMNode} from 'react-dom';
+
+export function assertRootElementContract(Component: React.ComponentType<any>): void {
+    describe('Root Element contract', function() {
+        const clientRenderer = new ClientRenderer();
+        function render<P>(element: React.ReactElement<P>): RenderingContext<P> & { node: Element } {
+            const output = clientRenderer.render(element);
+            return { node: findDOMNode(output.result as React.ReactInstance), ...output};
+        }
+
+        afterEach(() => {
+            clientRenderer.cleanup();
+        });
+
+        it('assumes data-automation-id merge', function() {
+             const {select} = render(<Component data-automation-id="CONTRACT_TEST"/>);
+             expect(select('CONTRACT_TEST'), 'data-automation-id not properly merged').to.not.equal(null);
```

Comment:
expect to select the root element specifically, no?
like `expect(node).to.have.attribute...`